### ISSUE TITLE
can no longer select text in fenced code blocks with style=console

### ISSUE THREAD

**Issue by wez** [squidfunk/mkdocs-material#6697]
### Context

The goal was to make the copy button copy only the "stuff I typed into the shell" portion of console fenced blocks. The CSS snippet shared here did facilitate that in the past. I'm not so interested in debugging this as I am in having a sustainable/supportable way of making that work in the future.

### Bug description

Seems to be a recent regression, but on pages like:
https://wezfurlong.org/wezterm/config/fonts.html#troubleshooting-fonts
the main portion of text in a fenced `console` highlight block is not selectable; it has variations on `user-select:none` in the CSS.

Page source can be found at:
https://raw.githubusercontent.com/wez/wezterm/main/docs/config/fonts.md

### Related links

https://squidfunk.github.io/mkdocs-material/reference/code-blocks/

seems like some "hacks" relating to this in the code:
https://github.com/search?q=repo%3Asquidfunk%2Fmkdocs-material%20user-select%3A%20none&type=code

### Reproduction

sorry, but no; seems like a trivial regression in the css somewhere

### Steps to reproduce

observe that it is not possible to select portions of the text shown on https://wezfurlong.org/wezterm/config/fonts.html#troubleshooting-fonts

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material/), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/).

**Comment by wez** [comment]
Digging into this more, I do have this in some custom CSS, so there is some egg on my face, however, this was working previously:

```css
/* prevent selection of prefix and output for console syntax */
.language-console .gp, .language-console .go {
  user-select: none;
  -webkit-user-select: none; /* Chrome/Safari */
  -ms-user-select: none; /* IE10+ */
  color: !important pink;
}
```

so it seems something about the shape of the markup/styles has changed unexpectedly.

Could you help me understand what changed and where?

**Comment by wez** [comment]
Perhaps a more productive way to look at this issue: the goal was to make the copy button copy only the "stuff I typed into the shell" portion of `console` fenced blocks. The snippet I shared above did facilitate that in the past.  I'm not so interested in debugging this as I am in having a sustainable/supportable way of making that work in the future.

**Comment by squidfunk** [comment]
Thanks for reporting.

> Reproduction
> sorry, but no; seems like a trivial regression in the css somewhere

Well, in that case, I'm afraid we can't help you, sorry. We need to precisely understand what worked before and ideally at what version it stopped working. It is only if you give us complete information, we can work towards a solution to your specific problem. [Closing as incomplete](https://squidfunk.github.io/mkdocs-material/contributing/#incomplete-issues).

**Comment by alexvoss** [comment]
Coming to this a bit late (been traveling) and possibly holding the wrong end of the stick but I had a look at the CSS on your site and could not find the CSS for doing this in `style.css`, which I assume is your extra CSS file? The other CSS included seems to be related to asciinema.

**Comment by wez** [comment]
@alexvoss I removed it; it was just that snippet in https://github.com/squidfunk/mkdocs-material/issues/6697#issuecomment-1910906655

Ignoring the framing of this particular issue as a bug/regression, my higher level purpose is for the copy button to copy only the the stuff that the user would type in a `console` highlight.

In this example, I would like the copy button to copy `ls -1` when clicked, as that is the thing that the user will want to copy and paste into their own terminal to duplicate the instructions on screen:

```console
$ ls -1
one
two
three
```

The CSS snippet, at the time it was written, worked together with https://github.com/wez/wezterm/blob/main/docs/javascript/fix-codeblock-console-copy-button.js which I think was cribbed from something I found in an internet search.

It's been a while since I looked at this stuff; skimming it now, it seems obvious that it would block user selection of the main body text and that probably that stuff should use some other technique to indicate which regions should be elided from the text.

Either way: I don't think there is anything for you to actively do anything about here, unless you'd like to integrate this concept as a first class feature in mkdocs-material :)
